src/OAuth/OAuth2/Service/BattleNetCN.php

```diff
@@ -0,0 +1,18 @@
+<?php
+namespace OAuth\OAuth2\Service;
+
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\Credentials;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+
+class BattleNetCN extends BattleNetBase
+{
+    public function __construct(Credentials $credentials, ClientInterface $httpClient, TokenStorageInterface $storage, $scopes = array(), UriInterface $baseApiUri = null)
```

Comment:
These create a lot of duplicate code (DRY: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). The logic is the same so you only need to keep the `BattleNet` service. The region and URI can be set on the created service. For example:

```php
$battleNet = $serviceFactory->createService(
        'BattleNet',
        $credentials,
        $storage,
        []
);

$battleNet->setRegion(BattleNet::REGION_CN);
```

Pick a region that makes sense to be the default. If none make sense you should throw an exception if someone tries to use the service without specifying a region.